Assets/Scripts/Game/GameManager.cs

```diff
@@ -1,12 +1,16 @@
 using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using Dialogue;
 
 public class GameManager : MonoBehaviour
 {
     private static GameManager instance;
     public static GameManager Instance { get { return instance; } }
 
+    public DialogueSystem DialogueSystem { get { return dialogueSystem; } }
+    [SerializeField] private DialogueSystem dialogueSystem;
```

Comment:
I would make this a direct static reference:
```suggestion
    private static DialogueSystem dialogueSystem;
    public static DialogueSystem DialogueSystem { get { return dialogueSystem; } }
```
Then use `GetComponent` in `Awake`:
```cs
dialogueSystem = GetComponent<DialogueSystem>();
```